### DISCUSSION TITLE
Error: Failure to open `simulation.msh' file

### DISCUSSION THREAD

**Discussion by Yuanguo1994** [neperfepx/FEPX#3] [Q&A] [answered]
Dear all,
 
I have difficult in installing fepx. Actually I have installed FEPX, OpenMPI and gcc. But when I input command fepx, some error occur as follows:

```
[i] Parsing file `simulation.config'...
Error  :     > Failure to open `simulation.msh' file.
```

I don’t know I made a mistake in which step. So could you please to help me.
By the way, I am using windows system with Ubuntu installed. Does this affects the work of fepx?

**Comment by jcappola** [comment, marked as answer]
The good news is that it seems you have installed FEPX successfully if you are getting that error message. That specific error message suggests that you are running FEPX _outside_ of a simulation directory. Please navigate to an example case provided with your installation (for example, within `/FEPX/examples/1_uniaxial`) and try running `fepx` in your terminal again. This should start a simulation proper if everything else is installed without issue.

We have never tested FEPX on a Windows system before, but if you have Ubuntu installed (either in a Virtual Machine or Dual Boot) you should be fine as long as you continue to work within Ubuntu.

EDIT: To be clear, `fepx` must be run _within_ a directory that contain _at minimum_ the files: `simulation.msh` and `simulation.config`.

**Reply by Yuanguo1994** [reply]
Thanks for your reply. But it fails again, as follows:

```
yt2n19@UOS-218102:/mnt/c/ubuntu-app/fepx-1.1.1$ cd examples/1_uniaxial
yt2n19@UOS-218102:/mnt/c/ubuntu-app/fepx-1.1.1/examples/1_uniaxial$ fepx
==========================    F   E   P   X   ==========================
Info   : A finite element software package for polycrystal plasticity.
Info   : Version 1.1.1
Info   : Running on 1 cores.
Info   : <https://fepx.info>
Info   : Copyright (C) 1996-2020, DPLab, ACME Lab.
Info   : ---------------------------------------------------------------
Info   : Start time: 2021-2-19 at 16:25
Info   : Loading simulation...
Info   :   [i] Parsing file `simulation.config'...
Error  :     > Failure to open `simulation.msh' file.
```

Is that problem about windows defender? Because when I input the  command and run,  the windows defender comes out and requires an access.

**Reply by mkasemer** [reply]
I agree with @jcappola: Your installation seems to be successful since FEPX is telling you what it is and is not parsing, and it is likely due to the fact that your folder does not contain the necessary files.

If you are running one of the examples, you must first copy over the `simulation.msh` file to the directory. Please view the specifics in the `runexample.sh` file below: note that it copies the necessary mesh file to the current simulation directory.

**EDIT:** In your above example, does the folder contain a file `simulation.msh`? Type `ls` to list the contents of the folder.

```
#!/bin/bash

# Run the `Uniaxial Control' example (`1_uniaxial') in FEPX
# - This bash script will copy the necessary mesh file from the external
#   directory `../mesh' by default. The mesh file may also be generated
#   using the provided `generate_mesh.sh' script if Neper is installed.
#
# - Executes FEPX with OpenMPI's `mpirun' on 2 processors by default.

# First, copy the mesh file into the directory. Comment this line if `generate_mesh.sh' is used.
cp ../mesh/simulation.msh .

# Then, run FEPX in parallel
mpirun -np 2 fepx

exit 0

# This script produces output files:
# - post.report
# - post.coo.*
# - post.strain-el.*
# - post.stress.*
# - post.force.*
```

**Reply by Yuanguo1994** [reply]
Thanks both. It works now!!!

Really thanks for your help

**Comment by sajutabraham** [comment]
I could install fepx successfully. But I am facing a problem while running the "generate_mesh.sh" given in the "1_uniaxial" example. The error message is "mesh partitioning are not available since this version of neper has built without libscotch". But when I tried installing libscotch, it shows already installed the newest version. What could be the reason behind this error? As a result "simulation.msh" is not generated but "simulation.tess" is generated. Can you please help?

**Reply by mkasemer** [reply]
@sajutabraham - what you are describing is a problem with your installation of Neper. Any problems with Neper should be posted to the Neper discussion forum [here](https://github.com/rquey/neper/discussions). This forum is for issues with FEPX.

To answer your question, you are unable to generate a mesh since the command used in `generate_mesh.sh` assumes your `neper` was built with `libscotch`. The error you are getting from Neper indicates that it was built without `libscotch` (you ). You must tell Neper to build with `libscotch` - simply having it installed to your system is not enough. Please post to the Neper discussion forum if you are still confused. You can see what your installation of `neper` was built against by reading the console output, for example:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.1.3-14
Info   : Built with: gsl|muparser|opengjk|nlopt|libscotch
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2020, and GNU GPL'd, by Romain Quey.
========================================================================
```

Alternatively, you can remove `-part X` from the meshing portion of `generate_mesh.sh`. This will allow you to generate a mesh without `libscotch`. However, your simulations will be inefficient if the mesh is not partitioned using `libscotch`, so in general this is not recommended.
